js/ui/status/keyboard.js

```diff
@@ -480,6 +481,41 @@ const InputSourceManager = new Lang.Class({
             this._updateMruSettings();
     },
 
+    passwordMode: function(enable) {
+        let source;
+
+        if (enable) {
+            // do not switch the leyboard layout if the current one is a latin one
+            if (this._keyboardManager.is_latin_layout(this._currentSource.id)) {
+                return;
+            }
+
+            // look for latin keyboard layout to switch to
+            for (let i = 0; i < this._mruSources.length; ++i) {
+                let src = this._mruSources[i];
+                if (this._keyboardManager.is_latin_layout(src.id)) {
+                    source = src;
+                    break;
+                }
+            }
```

Comment:
Assuming that we have ES6 suport in Gjs, you can express this as

    source = this._mruSources.find(src => this._keyboardManager.is_latinlayout(src.id)) || new InputSource('xkb', 'us', 'English (US)', 'us', 0);